src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java

```diff
@@ -0,0 +1,253 @@
+import io.sphere.sdk.producttypes.queries.ProductTypeQuery;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.queries.QueryPredicate;
+import io.sphere.sdk.types.ResourceTypeIdsSetBuilder;
+import io.sphere.sdk.types.Type;
+import io.sphere.sdk.types.TypeDraft;
+import io.sphere.sdk.types.TypeDraftBuilder;
+import io.sphere.sdk.types.commands.TypeCreateCommand;
+import io.sphere.sdk.types.commands.TypeDeleteCommand;
+import io.sphere.sdk.types.queries.TypeQuery;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import javax.annotation.Nonnull;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import uk.org.lidalia.slf4jtest.TestLogger;
+import uk.org.lidalia.slf4jtest.TestLoggerFactory;
+
+class CliRunnerIT {
+  private static final TestLogger testLogger = TestLoggerFactory.getTestLogger(CliRunner.class);
+  private static ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+  private static PrintStream originalSystemOut;
+
+  @BeforeAll
+  static void setupSuite() throws UnsupportedEncodingException {
+    final PrintStream printStream = new PrintStream(outputStream, false, "UTF-8");
+    originalSystemOut = System.out;
+    System.setOut(printStream);
+
+    deleteTestData(createClient(CTP_SOURCE_CLIENT_CONFIG), createClient(CTP_TARGET_CLIENT_CONFIG));
+  }
+
+  private static void deleteTestData(
+      @Nonnull final SphereClient sourceClient, @Nonnull final SphereClient targetClient) {
+
+    deleteTestData(sourceClient);
+    deleteTestData(targetClient);
+  }
+
+  static void deleteTestData(@Nonnull final SphereClient client) {
+    queryAndExecute(client, CategoryQuery.of(), CategoryDeleteCommand::of);
+    queryAndExecute(client, ProductQuery.of(), ProductDeleteCommand::of);
+    queryAndExecute(client, ProductTypeQuery.of(), ProductTypeDeleteCommand::of);
+    queryAndExecute(client, TypeQuery.of(), TypeDeleteCommand::of);
+    queryAndExecute(client, InventoryEntryQuery.of(), InventoryEntryDeleteCommand::of);
+  }
+
+  @AfterAll
+  static void tearDownSuite() {
+    System.setOut(originalSystemOut);
+  }
+
+  @AfterEach
+  void tearDownTest() {
+    testLogger.clearAll();
+  }
+
+  @Test
+  void run_WithSyncAsArgumentWithAllArg_ShouldExecuteAllSyncers()
+      throws UnsupportedEncodingException {
+    // preparation
+    final String resourceKey = "foo";
+    try (final SphereClient targetClient = createClient(CTP_TARGET_CLIENT_CONFIG)) {
+      try (final SphereClient sourceClient = createClient(CTP_SOURCE_CLIENT_CONFIG)) {
+        setupTestData(sourceClient, resourceKey);
+
+        final SyncerFactory syncerFactory = SyncerFactory.of(sourceClient, targetClient);
+
+        // test
+        CliRunner.of()
+            .run(new String[] {"-s", "all"}, () -> syncerFactory)
```

Comment:
Just an idea:

what if we return an error if `Options` is longer than 2 elements?

Reply:
Could be an improvement.  Will create an issue for it that could be addressed later.

Reply:
https://github.com/commercetools/commercetools-project-sync/issues/13